apps/examples/stdk_smart_switch/smart_switch.c

```diff
@@ -0,0 +1,304 @@
+/* ***************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+//for implementing main features of IoT device
+#include <stdbool.h>
+#include <stdio.h>
+#include "st_dev.h"
+#include "device_control.h"
+#include "device_info.h"
+#include "onboarding_config.h"
+#include "iot_os_util.h"
+
+static const char *TAG = "SLamp";
+#define SL_LOGE(tag, format, ...) do {printf("%s ", tag); printf(format"\n", ##__VA_ARGS__); } while (0)
+#define SL_LOGI(tag, format, ...) do {printf("%s ", tag); printf(format"\n", ##__VA_ARGS__); } while (0)
+
+enum smartlamp_switch_onoff_state {
+	SMARTLAMP_SWITCH_OFF = 0,
+	SMARTLAMP_SWITCH_ON = 1,
+};
+
+static int smartlamp_switch_state = SMARTLAMP_SWITCH_ON;
+
+static iot_status_t g_iot_status;
+
+static int noti_led_mode = LED_ANIMATION_MODE_IDLE;
+
+/* TODO: Need 'get_ctx' function (get ctx from cap_handle) */
+IOT_CTX* ctx = NULL;
+
+static void change_switch_state(int32_t state)
+{
+	/* change state */
+	smartlamp_switch_state = state; 
+	if (state == SMARTLAMP_SWITCH_ON) {
+		gpio_set_level(GPIO_OUTPUT_NOTIFICATION_LED, NOTIFICATION_LED_GPIO_ON);
+		gpio_set_level(GPIO_OUTPUT_MAINLED, MAINLED_GPIO_ON);
+	} else {
+		gpio_set_level(GPIO_OUTPUT_NOTIFICATION_LED, NOTIFICATION_LED_GPIO_OFF);
+		gpio_set_level(GPIO_OUTPUT_MAINLED, MAINLED_GPIO_OFF);
+	}
+}
+
+static void send_switch_cap_evt(IOT_CAP_HANDLE *handle, int32_t state)
+{
+	IOT_EVENT *switch_evt;
+	uint8_t evt_num = 1;
+	int32_t sequence_no;
+
+	/* Setup switch onoff state */
+	if (state == SMARTLAMP_SWITCH_ON) {
+		switch_evt = st_cap_attr_create_string("switch", "on", NULL);
+	} else {
+		switch_evt = st_cap_attr_create_string("switch", "off", NULL);
+	}
+
+	/* Send switch onoff event */
+	sequence_no = st_cap_attr_send(handle, evt_num, &switch_evt);
+	if (sequence_no < 0) {
+		SL_LOGE(TAG, "fail to send switch onoff data");
+	}
+
+	SL_LOGI(TAG, "Sequence number return : %d", sequence_no);
+	st_cap_attr_free(switch_evt);
+}
+
+static void button_event(IOT_CAP_HANDLE *handle, uint32_t type, uint32_t count)
+{
```

Comment:
Need to check whether handle is NULL or not.

Reply:
fixed